snapcraft.yaml

```diff
@@ -43,6 +46,9 @@ apps:
       PYTHON: $SNAP/usr/bin/python3.10
       PY3OR2_PYTHON: $SNAP/usr/bin/python3.10
       PATH: $PATH:$SNAP/bin:$SNAP/sbin
+      # APPORT_DATA_DIR must be set before the `apport` python module is
+      # imported.
+      APPORT_DATA_DIR: $SNAP/share/apport
```

Comment:
```suggestion
      # APPORT_DATA_DIR must be set before the `apport` python module is
      # imported.
      APPORT_DATA_DIR_ORIG: $APPORT_DATA_DIR
      APPORT_DATA_DIR: $SNAP/share/apport
```

Reply:
I think if we add an _ORIG here for APPORT_DATA_DIR this would fix running `ubuntu-bug subiquity` in the debug shell too.

Reply:
@Chris-Peterson444 in the debug shell do you want the out-of-snap behavior or the in-the-snap behavior?

Reply:
out-of-snap for now. I don't really have any hard opinions on either way yet, but my understanding of the current intent of the debug shell behavior is that it should perform like an out-of-snap shell ( i.e., it should be pretty close to what you could get by sshing into the live machine) so we should be consistent. Currently APPORT_DATA_DIR pointing to the snap path makes this sorta not true. Although, I can see the merits for eventually pointing this back to the snap path and having some apport hook we could update with a snap refresh.

Reply:
I did this change. But now when I run `ubuntu-bug subiquity` from the debug shell, I'm asked to choose between:
```
*** You have two versions of this application installed, which one do you want to report a bug against?

Choices:
  1: subiquity 24.10-devel+git213.443b8f43 () snap
  2: subiquity (x1 of snap /snap/subiquity/x1 of version 24.10-devel+git213.443b8f43) deb package
  C: Cancel
```

(yes, I injected an updated snap). 

Not sure if I should push this change yet?

Reply:
Maybe but it does not happen with an updated snap without the APPORT_DATA_DIR_ORIG variable. So I'm not sure, @Chris-Peterson444 what do you think?

Reply:
That looks like the right prompt to me. I can't remember the specifics of how apport gets the first option on server, but either choice will still run the package hook (choosing "snap" just also appends some SnapInfo fields). 

The current behavior is that the "deb package" is not found since `$SNAP/usr/share/apport/package-hooks/subiquity.py` doesn't exist. So I believe it's some default snap info collection that runs otherwise.

Reply:
ACK. Thanks for the feedback. Merging with the change then.